Snakefile

```diff
@@ -0,0 +1,94 @@
+import pandas as pd
+import sourmash
+import os
+
+# create wildcard variables for workflow
+metadata = pd.read_csv("inputs/metadata-paired-mgx-mtx.tsv", sep = "\t") # read in metadata as a pandas dataframe
+MTX = metadata['mtx_run_accession'].unique().tolist()                    # make run accession in mtx col into list
+MGX = metadata['mgx_run_accession'].unique().tolist()                    # make run accession in mgx col into list
+RUN_ACCESSIONS = MGX + MTX                                               # combine mtx and mgx into one list with all run accessions
+SAMPLES = metadata['sample_name'].unique().tolist()                      # make a list of sample names
+KSIZES = [21, 31, 51]                                                    # create a list of k-mer sizes for the workflow
+
+rule all:
+    input: expand("outputs/sourmash_sketch_subtract/{sample}_k{ksize}.sig", sample = SAMPLES, ksize = KSIZES)
+
+rule sourmash_sketch:
+    """
+    Use fastq-dump to download sequencing data for a given accession.
+    Pipe the sequencing data directly to a sketch without writing to disk.
+    Using fastq-dump like this dumps both single end and paired end reads into std out, which can then be piped into sourmash sketch
+    """
+    output: "outputs/sourmash_sketch/{run_accession}.sig"
+    conda: 'envs/sourmash.yml'
+    shell:'''
+    fastq-dump --disable-multithreading --fasta 0 --skip-technical --readids --read-filter pass --dumpbase --split-spot --clip -Z {wildcards.run_accession} | 
+        sourmash sketch dna -p k=21,k=31,k=51,scaled=200,abund --name {wildcards.run_accession} -o {output} -
+    '''
+
+rule calculate_mtx_not_in_mgx:
+    """
+    Use the sourmash python API to subtract metagenome fracminhash sketches from metatranscriptome fracminhash sketches.
+    This for loop re-implements a simplified veresion of the the CLI of sourmash sig subtract.
+    https://github.com/sourmash-bio/sourmash/blob/latest/src/sourmash/sig/__main__.py
+    Doing this in a python script is the easiest way to map pairs of samples to each other within the snakemake workflow.
+    
+    The run directive doesn't allow a run to have a conda environment; 
+    therefore the run environment for this code is specified in environment.yml
+    """
+    input:
+        metadata = 'inputs/metadata-paired-mgx-mtx.tsv',
+        sigs = expand('outputs/sourmash_sketch/{run_accession}.sig', run_accession = RUN_ACCESSIONS) 
+    output:
+        sigs = expand("outputs/sourmash_sketch_subtract/{sample}_k{{ksize}}.sig", sample = SAMPLES)
+    run:
+        ksize = wildcards.ksize
+        # read in metadata dataframe to derive sample pairs
+        metadata = pd.read_csv(input.metadata, sep = "\t") # read in metadata as pandas df
```

Comment:
haven't we already read this at the beginning of the file?

Reply:
is there a reason to re-read it?

Reply:
yes we have...I'm not sure if the `run:` invocations within rules inherit global variables...I've always assumed they don't but I should probably test whether that's actually true

Reply:
ok can confirm that it uses global variables, so there is in fact no reason to re-read it in here since no operations have modified it since it was read in at the top of the file. Thank you!

Reply:
```suggestion
```